notebooks/Benchmark_notebook.ipynb

```diff
@@ -54,20 +54,33 @@
    "outputs": [],
    "source": [
     "e = 2.7182818284590452353602874 #hardcoded to avoid dependence on torch / numpy\n",
+    "runtime_f = 0 #Runtime global variable\n",
     "def test_function_1d(x):\n",
     "    \"\"\" As a test function in 1D we use e^x\n",
     "    \"\"\"\n",
-    "    return e**x\n",
+    "    global runtime_f\n",
```

Comment:
I don't get the point of using a global variable if you overwrite it every time? Wouldn't it be better to accumulate all function runtimes with

```python
start_f = perf_counter()
...
global runtime_f
runtime_f = runtime_f + (perf_counter() - start)
```

and then take that value at the end?

Reply:
Further, maybe you can create a function

```python
def track_fn_runtime(start):
global runtime_f
runtime_f = runtime_f + (perf_counter() - start)
```

which improves readability and saves you a few lines?